### PR TITLE
Add row number to leaderboard #1297

### PR DESCRIPTION
**Pull request by Simon-Busch**
- [x] Add raking to wardens data 
- [x] Update leaderboard table component
- [x] Update styling
- [x] Update mobile styling

### CODE REVIEW

**Comment by Simon-Busch** [comment]
Feature is done, @captainmangoC4  check below a screenshot and let me know if OK for you :-) 
Capture d’écran 2022-05-05 à 08 29 06

File: src/components/LeaderboardTable.js [hunk 1]
```diff
@@ -6,6 +6,14 @@ import LeaderboardHandle from "./LeaderboardHandle";
 const LeaderboardTable = ({ results }) => {
   const columns = React.useMemo(
     () => [
+      {
+        Header: 'Rank',
+        Cell: (props) => {
+          return (
+            <p>{props.row.original.rank}.</p>
+          )
```
Comment (joshuashort) [inline src/components/LeaderboardTable.js:14]: Sorry for the nit-picky edits; it's a good time to point out some styles that we've been adopting.

Like semi-colons.

```suggestion
          );
```

File: src/pages/leaderboard.js [hunk 2]
```diff
@@ -89,7 +89,7 @@ const Leaderboard = ({ data }) => {
   const handles = data.handles.edges;
 
   const resultData = useMemo(() => {
-    const result = [];
+    let result = [];
```
Comment (joshuashort) [inline src/pages/leaderboard.js:92]: Fair enough, but re-writing the return statement might be better in this case.

```suggestion
    const result = [];
```

File: src/pages/leaderboard.js [hunk 3]
```diff
@@ -108,6 +108,7 @@ const Leaderboard = ({ data }) => {
         gasOptz: 0,
         allFindings: 0,
         awardTotal: 0,
+        rank:0
```
Comment (joshuashort) [inline src/pages/leaderboard.js:111]: Usually not so compact, and with trailing commas (when supported)

```suggestion
        rank: 0,
```

File: src/pages/leaderboard.js [hunk 4]
```diff
@@ -120,6 +121,14 @@ const Leaderboard = ({ data }) => {
...
     }
+    result.sort((a,b) => {
+      if (a.awardTotal > b.awardTotal) return -1;
+      if (b.awardTotal < a.awardTotal) return 1;
+      return 0;
+    })
+    result = result.map((element, index) => {
+      return {...element, rank: index+1 }
+    })
     return result;
```
Comment (joshuashort) [inline src/pages/leaderboard.js:132]:
```suggestion

    return result
      .sort((a,b) => {
        return b.awardTotal - a.awardTotal;
      })
      .map((element, index) => {
        return { ...element, rank: index + 1 };
      });
```

File: src/styles/leaderboard.css [hunk 5]
```diff
@@ -302,4 +313,4 @@
   .leaderboard-table td:nth-of-type(8):before {
     content: "GAS";
   }
-}
+}
```
Comment (joshuashort) [inline src/styles/leaderboard.css:316]: In general, ending with a newline

```suggestion
}

```

**Comment by sockdrawermoney** [comment]
Have had several people give of feedback on the current mobile styles for the table, so I am glad to see that getting revisited. 

While the current approach does a good job of surfacing information, it breaks the context of rankings.

A site which I think does an exceptional job of showing stats tables on mobile is Fangraphs. The header column and header row are pinned and the table data scrolls the overflow within the mobile view without needing horizontal scrolling of the page.

See tables on this page: https://www.fangraphs.com/players/ty-france/17982/stats?position=1B

I also think on mobile we should not show team member listings; maybe they become a popover when you click an (i) icon or something next to the team name.

**Comment by Simon-Busch** [comment]
> Have had several people give of feedback on the current mobile styles for the table, so I am glad to see that getting revisited.
> 
> While the current approach does a good job of surfacing information, it breaks the context of rankings.
> 
> A site which I think does an exceptional job of showing stats tables on mobile is Fangraphs. The header column and header row are pinned and the table data scrolls the overflow within the mobile view without needing horizontal scrolling of the page.
> 
> See tables on this page: https://www.fangraphs.com/players/ty-france/17982/stats?position=1B
> 
> I also think on mobile we should not show team member listings; maybe they become a popover when you click an (i) icon or something next to the team name.


I am trying to figure things out to have a clean interface on the table on mobile.

Actually, the table on. fangraphs are a bit packed a don't "breath" a lot if I can say it this way.

We could imagine something lighter like:

| Rank      | Warden  | Total Award | 
| ----------- | ----------- | ----------- |
| 1      | XXXX       | 99999 |
| 2      | XXXX       | 99999 |
....

An **on click** being redirected to a Warden page with more detailed statistics || informations.
-> all data from the leader bord
-> contests in which he participated
-> ... many more

What would you think about it @sockdrawermoney  @tealtigrrr  ?

**Comment by JustDravee** [comment]
Hey frens! 
I feel like this PR contains a lot of things (mobile and web together, with further questioning and such on detailled stats).
Wouldn't it be more agile to ship the working stuff and make smaller PRs? (I guess here, everything except mobile?). Then ship the mobile styling and stats stuff later?
Just my 0.02 USDC (2 cents :p)

**Comment by sockdrawermoney** [comment]
Yes, agree with @JustDravee here that mobile changes don't need to be in this PR. I was just awake in the middle of the night looking at stuff and saw this mention mobile and dropped the note in that context. 

Also @Simon-Busch to your question I'm not referring to the aesthetic of the fangraphs tables but the specific method they use to overflow table width on mobile.

**Comment by Simon-Busch** [comment]
Right, totally make sense, will open later a different PR for the next steps :) 👍🏼

File: src/components/LeaderboardTable.js [hunk 6]
```diff
@@ -6,6 +6,14 @@ import LeaderboardHandle from "./LeaderboardHandle";
...
   const columns = React.useMemo(
     () => [
+      {
+        Header: 'Rank',
+        Cell: (props) => {
+          return (
+            <p>{props.row.original.rank}.</p>
+          );
+        }
+      },
```
Comment (joshuashort) [inline src/components/LeaderboardTable.js:16]: Here's simple row numbering.

```suggestion
      {
        Header: '#',
        Cell: (props) => {
          return (
            <p>{props.flatRows.indexOf(props.row) + 1}</p>
          );
        }
      },
```

Comment (captainmangoC4) [reply src/components/LeaderboardTable.js:16]: Clever!

File: src/pages/leaderboard.js [hunk 7]
```diff
@@ -120,6 +120,7 @@ const Leaderboard = ({ data }) => {
         result.push(combinedData);
       }
     }
+
```
Comment (joshuashort) [inline src/pages/leaderboard.js:123]:
```suggestion
```

**Review by joshuashort — APPROVED** [review]

**Pull request merged**